src/sentry/utils/snuba.py

```diff
@@ -797,20 +806,21 @@ def _aliased_query_impl(
         raise ValueError("A dataset is required, and is no longer automatically detected.")
 
     derived_columns = []
+    resolve_func = resolve_column(dataset)
     if selected_columns:
         for (i, col) in enumerate(selected_columns):
             if isinstance(col, (list, tuple)):
                 derived_columns.append(col[2])
             else:
-                selected_columns[i] = resolve_column(col, dataset)
+                selected_columns[i] = resolve_func(col)
         selected_columns = [c for c in selected_columns if c]
 
     if aggregations:
         for aggregation in aggregations:
             derived_columns.append(aggregation[2])
 
     if conditions:
-        condition_resolver = condition_resolver or resolve_column
+        condition_resolver = condition_resolver or resolve_func
```

Comment:
Does this work if `condition_resolver == resolve_func`? Looks like we pass `dataset` to it.

Reply:
Nope, it doesn't. Tests were yelling about that.